tests/bud.bats

```diff
@@ -534,6 +534,114 @@ _EOF
   assert "$output" !~ "unwanted stage"
 }
 
+
+# Verify --mount=type=cache with additional option U.
+# Following test ensures that a common cache created
+# with different ownership is accessible with additional
+# option.
+@test "build with --mount=type=cache and test ownership" {
+  mkdir -p ${TEST_SCRATCH_DIR}/bud/platform
+  echo something > ${TEST_SCRATCH_DIR}/bud/platform/somefile
+
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile << _EOF
+FROM alpine
+COPY somefile .
+RUN --mount=type=cache,target=/mycache,z cp somefile /mycache/somefile
+RUN --mount=type=cache,target=/mycache,z mkdir -p /mycache/foo/bar
+RUN --mount=type=cache,target=/mycache,z cat /mycache/somefile
+_EOF
+
+  # build should be successful
+  run_buildah build $WITH_POLICY_JSON -t source -f ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile ${TEST_SCRATCH_DIR}/bud/platform
+  expect_output --substring "something"
+
+
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile << _EOF
+FROM alpine
+COPY somefile .
+RUN addgroup -S appgroup && adduser -S appuser -G appgroup
+USER 100:101
+RUN --mount=type=cache,target=/mycache,z,U cp somefile /mycache/somefile
+RUN --mount=type=cache,target=/mycache,z,U mkdir -p /mycache/foo/bar
+RUN --mount=type=cache,target=/mycache,z,U cat /mycache/somefile
+_EOF
+
+  # build should be successful since `U` is used
+  run_buildah build $WITH_POLICY_JSON -t source -f ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile ${TEST_SCRATCH_DIR}/bud/platform
+  expect_output --substring "something"
+
+  # Test with custom uid and gid explicitly instead of `U`
+  cat > ${TEST_SCRATCH_DIR}/bud/platform/Dockerfile << _EOF
```

Comment:
Can you verify the final file is owned by the specified user?

Reply:
Yes amended the test to check output of `ls -lrt /mycache/somefile`